app/models/manageiq/providers/openstack/inventory/parser/cloud_manager.rb

```diff
@@ -205,9 +205,29 @@ def miq_templates
       operating_system.product_name = guest_os
       operating_system.distribution = i.try(:os_distro)
       operating_system.version = i.try(:os_version)
+
+      if snapshot?(i) && parent_server_uid
+        snapshot = persister.snapshots.find_or_build(i.id)
+        snapshot.name = i.name
+        snapshot.uid  = i.id
+        snapshot.uid_ems = i.id
+        snapshot.ems_ref = i.id
+        snapshot.create_time = i.created_at
+        snapshot.description = i.attributes[:description]
+        snapshot.vm_or_template = persister.vms.lazy_find(parent_server_uid)
+      end
     end
   end
 
+  def snapshot?(image)
+    block_device_mapping = image.attributes[:block_device_mapping]
+    source_type = JSON.parse(block_device_mapping).try(:first).try(:[], 'source_type') if block_device_mapping
+
+    (image.image_type == 'snapshot') || (source_type == 'snapshot')
```

Comment:
How about `return true if image.image_type == 'snapshot'` early to avoid having to  `JSON.parse()` if the first condition of your `||` is true?